kornia/losses/ms_ssim_l1.py

```diff
@@ -0,0 +1,164 @@
+    ) -> torch.Tensor:
+        """Create 1-D gauss kernel.
+
+        Args:
+            size: the size of gauss kernel.
+            sigma: sigma of normal distribution.
+
+        Returns:
+            1D kernel (size).
+        """
+        coords = torch.arange(size, device=device, dtype=dtype)
+        coords -= size // 2
+        g = torch.exp(-(coords ** 2) / (2 * sigma ** 2))
+        g /= g.sum()
+        return g.reshape(-1)
+
+    def _fspecial_gauss_2d(
+        self, size: int, sigma: float, device: Optional[torch.device] = None, dtype: Optional[torch.dtype] = None
+    ) -> torch.Tensor:
+        """Create 2-D gauss kernel.
+
+        Args:
+            size: the size of gauss kernel.
+            sigma: sigma of normal distribution.
+
+        Returns:
+            2D kernel (size x size).
+        """
+        gaussian_vec = self._fspecial_gauss_1d(size, sigma, device, dtype)
+        return torch.outer(gaussian_vec, gaussian_vec)
+
+    def forward(self, img1: torch.Tensor, img2: torch.Tensor) -> torch.Tensor:
+        """Compute MS_SSIM loss.
+
+        Args:
+            img1: the predicted image with shape :math:`(B, C, H, W)`.
+            img2: the target image with a shape of :math:`(B, C, H, W)`.
+
+        Returns:
+            Estimated MS-SSIM_L1 loss.
+        """
+        if not isinstance(img1, torch.Tensor):
+            raise TypeError(f"Input type is not a torch.Tensor. Got {type(img1)}")
+
+        if not isinstance(img2, torch.Tensor):
+            raise TypeError(f"Output type is not a torch.Tensor. Got {type(img2)}")
+
+        if not len(img1.shape) == len(img2.shape):
+            raise ValueError(f"Input shapes should be same. Got {type(img1)} and {type(img2)}.")
+
+        g_masks: torch.Tensor = torch.jit.annotate(torch.Tensor, self._g_masks)
+
+        CH: int = img1.shape[-3]
+        mux = F.conv2d(img1, g_masks, groups=CH, padding=self.pad)
+        muy = F.conv2d(img2, g_masks, groups=CH, padding=self.pad)
+        mux2 = mux * mux
+        muy2 = muy * muy
+        muxy = mux * muy
+
+        sigmax2 = F.conv2d(img1 * img1, g_masks, groups=CH, padding=self.pad) - mux2
+        sigmay2 = F.conv2d(img2 * img2, g_masks, groups=CH, padding=self.pad) - muy2
+        sigmaxy = F.conv2d(img1 * img2, g_masks, groups=CH, padding=self.pad) - muxy
+
+        lc = (2 * muxy + self.C1) / (mux2 + muy2 + self.C1)
+        cs = (2 * sigmaxy + self.C2) / (sigmax2 + sigmay2 + self.C2)
+        lM = lc[:, -1, :, :] * lc[:, -2, :, :] * lc[:, -3, :, :]
+        PIcs = cs.prod(dim=1)
+
+        # Compute MS-SSIM loss
+        loss_ms_ssim = 1 - lM * PIcs
+
+        # TODO: pass pointer to function e.g. to make more custom with mse, cosine, etc.
+        # Compute L1 loss
+        loss_l1 = F.l1_loss(img1, img2, reduction='none')
+
+        # Compute average l1 loss in 3 channels
+        gaussian_l1 = F.conv2d(loss_l1, g_masks.narrow(dim=0, start=-CH, length=CH), groups=CH, padding=self.pad).mean(
```

Comment:
```suggestion
        gaussian_l1 = F.conv2d(loss_l1, g_masks[-CH:], groups=CH, padding=self.pad).mean(
```
@hasibzunair I think this equivalent

Reply:
changed in new commit